app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkChaptersCompletedFragmentTest.kt

```diff
@@ -422,6 +423,58 @@ class MarkChaptersCompletedFragmentTest {
     }
   }
 
+  @Test
+  fun testMarkChaptersCompletedFragment_unSelectAll_unSelectsAllChapters() {
+    launch<MarkChaptersCompletedTestActivity>(
+      createMarkChaptersCompletedTestActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.mark_chapters_completed_all_check_box_container)).perform(click())
+      scrollToPosition(position = 1)
+      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 1)
+      scrollToPosition(position = 2)
+      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 2)
+      scrollToPosition(position = 3)
+      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 3)
+      scrollToPosition(position = 5)
+      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 5)
+      scrollToPosition(position = 7)
+      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 7)
+      scrollToPosition(position = 8)
+      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 8)
+      scrollToPosition(position = 10)
+      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 10)
+      scrollToPosition(position = 11)
+      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 11)
+      scrollToPosition(position = 13)
+      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 13)
+      scrollToPosition(position = 14)
+      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 14)
+      onView(withId(R.id.mark_chapters_completed_all_check_box_container)).perform(click())
```

Comment:
It seems like this test is verifying multiple things. I suggest splitting it up to keep the tests focused:
- One test for verifying that clicking the 'check all' does, indeed, check all items
- One test for verifying that checking all, then unchecking them, does indeed uncheck all items

Reply:
Actually @BenHenning i am verifying if unchecking them does indeed uncheck and also the items as this is the issue at hand but at first i have to check the checkbox and all items i have added the check logic into a separate function in a separate function to avoid duplicating. PLMK if this works.

Reply:
I'm not sure that the verify step is needed, though. If there's already a test to verify that clicking the 'check all' box checks everything, then your test can start with that assumption by clicking the checkbox, then unclicking it and verifying that nothing is checked anymore (both tests together tell the whole story).

Reply:
Have started with assumption and updated the tests accordingly.